### PR TITLE
feat(rule): heading element should have content

### PR DESCRIPTION
**Pull request by mohammedzamakhan**
https://dequeuniversity.com/rules/axe/1.1/empty-heading

### CODE REVIEW

File: src/templateAccessibilityHeadingContentRule.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        start: { offset: startOffset }
+      }
+    } = element;
+    this.addFailureFromStartToEnd(startOffset, endOffset, Rule.FAILURE_MESSAGE);
+  }
+}
+
+export class Rule extends Rules.AbstractRule {
+  static readonly metadata: IRuleMetadata = {
+    description: 'Ensures that the heading element has some content in it',
```
Comment (mgechev) [inline src/templateAccessibilityHeadingContentRule.ts]: You can drop some

**Comment by mgechev** [comment]
It makes sense to merge this rule with the anchor one, same functionality?

File: src/templateAccessibilityHeadingContentRule.ts (hunk 1 above)
Comment (mohammedzamakhan) [reply src/templateAccessibilityHeadingContentRule.ts]: Done

**Comment by mgechev** [comment]
@mohammedzamakhan looks good! Would you fix the failures and we can merge!

**Comment by mohammedzamakhan** [comment]
@mgechev Done

File: src/templateAccessibilityElementsContentRule.ts [hunk 2]
```diff
@@ -32,21 +32,22 @@ class TemplateAccessibilityAnchorContentVisitor extends BasicTemplateAstVisitor
...
-    rationale: 'Anchor elements should have content to be accessible by screen readers',
-    ruleName: 'template-accessibility-anchor-content',
+    rationale: 'Heading and anchor elements should have content to be accessible by screen readers',
+    ruleName: 'template-accessibility-elements-content',
     type: 'functionality',
     typescriptOnly: true
   };
 
-  static readonly FAILURE_MESSAGE = 'Anchor element should have content';
+  static readonly FAILURE_MESSAGE = 'Heading elements should have content';
```
Comment (wKoza) [inline src/templateAccessibilityElementsContentRule.ts]: Heading or Anchor

File: src/templateAccessibilityElementsContentRule.ts [hunk 3]
```diff
@@ -32,21 +32,22 @@ class TemplateAccessibilityAnchorContentVisitor extends BasicTemplateAstVisitor
 
 export class Rule extends Rules.AbstractRule {
   static readonly metadata: IRuleMetadata = {
-    description: 'Ensures that the anchor element has some content in it',
+    description: 'Ensures that the heading and anchor elements have content in it',
```
Comment (wKoza) [inline src/templateAccessibilityElementsContentRule.ts]: In the same way, I see that button have a [similar need](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role#Accessibility_concerns).
Can we again broaden the rule ?

File: src/templateAccessibilityElementsContentRule.ts (hunk 2 above)
Comment (mgechev) [reply src/templateAccessibilityElementsContentRule.ts]: @mohammedzamakhan would you address this comment?

Comment (mohammedzamakhan) [reply src/templateAccessibilityElementsContentRule.ts]: Done 👍

File: src/templateAccessibilityElementsContentRule.ts (hunk 3 above)
Comment (mohammedzamakhan) [reply src/templateAccessibilityElementsContentRule.ts]: Done

File: test/templateAccessibilityElementsContentRule.spec.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        class Bar {}
+      `;
+      assertAnnotated({
+        message: getErrorMessage('a'),
+        ruleName,
+        source
+      });
+    });
+
+    it('should fail with no content in anchor tag', () => {
```
Comment (wKoza) [inline test/templateAccessibilityElementsContentRule.spec.ts]: the title isn't correct

File: test/templateAccessibilityElementsContentRule.spec.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        @Component({
+          template: \`
+            <h1>Heading Content!</h1>
+            <h2><app-content></app-content></h2>
+            <h3 [innerHTML]="dangerouslySetHTML"></h3>
+            <h4 [innerText]="text"></h4>
+            <a>Anchor Content!</a>
+            <a><app-content></app-content></a>
+            <a [innerHTML]="dangerouslySetHTML"></a>
+            <a [innerText]="text"></a>
```
Comment (wKoza) [inline test/templateAccessibilityElementsContentRule.spec.ts]: can you add a test with the `button`

**Comment by mgechev** [comment]
Let's merge https://github.com/mgechev/codelyzer/pull/754 first.

**Comment by mohammedzamakhan** [comment]
@mgechev can you merge this?

**Pull request merged**